### PR TITLE
[rimraf] Remove types

### PR DESCRIPTION
**Pull request by ffflorian**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If removing a declaration:
- [x] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

rimraf's version [4.0](https://github.com/isaacs/rimraf/blob/main/CHANGELOG.md#v40) now ships types directly.

@peterblazejewicz I am not 100% sure if this is the right way to remove the whole package - what happens if a new v3 version gets published? Should we keep the types for v2 and v3 maybe?

### CODE REVIEW

File: types/vinyl-fs/package.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "private": true,
+  "dependencies": {
+      "rimraf": "^4.0.0"
```
Comment (peterblazejewicz) [inline types/vinyl-fs/package.json:4]: this is probably not right,
what's the version of rimraf used by native vinyl-fs module? The version 4.0 introduce modules (ESM/CJS)

**Comment by peterblazejewicz** [comment]
> what happens if a new v3 version gets published? Should we keep the types for v2 and v3 maybe?

I think v3 was never bumped on DT, as the rimraf v2 > v3 were only minor changes without api changes (I could be wrong thougt), so no version bump on DT. This is unlikely v3 gets updates (3 years old version, really)

File: types/vinyl-fs/package.json (hunk 1 above)
Comment (ffflorian) [reply types/vinyl-fs/package.json:4]: You are right!

**Comment by ffflorian** [comment]
We are finally green 😃

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@ffflorian thanks!

**Comment by ffflorian** [comment]
Ready to merge

**Pull request merged**